zk_toolbox/crates/zk_inception/src/commands/prover/args/init.rs

```diff
@@ -0,0 +1,390 @@
+            setup_key_path,
+        }
+    }
+
+    fn partial_create_gcs_bucket_config_provided(
+        &self,
+        bucket_name: Option<String>,
+        location: Option<String>,
+        project_id: Option<String>,
+    ) -> bool {
+        bucket_name.is_some() || location.is_some() || project_id.is_some()
+    }
+
+    fn partial_gcs_config_provided(
+        &self,
+        bucket_base_url: Option<String>,
+        credentials_file: Option<String>,
+    ) -> bool {
+        bucket_base_url.is_some() || credentials_file.is_some()
+    }
+
+    fn handle_file_backed_config(&self, store_dir: Option<String>) -> ProofStorageConfig {
+        let proof_store_dir = store_dir.unwrap_or_else(|| {
+            Prompt::new(MSG_PROOF_STORE_DIR_PROMPT)
+                .default(DEFAULT_PROOF_STORE_DIR)
+                .ask()
+        });
+
+        ProofStorageConfig::FileBacked(ProofStorageFileBacked { proof_store_dir })
+    }
+
+    fn handle_gcs_config(
+        &self,
+        project_ids: Vec<String>,
+        bucket_base_url: Option<String>,
+        credentials_file: Option<String>,
+    ) -> ProofStorageConfig {
+        if !self.partial_gcs_config_provided(bucket_base_url.clone(), credentials_file.clone()) {
+            if PromptConfirm::new(MSG_CREATE_GCS_BUCKET_PROMPT).ask() {
+                return self.handle_create_gcs_bucket(project_ids, None, None, None, None);
+            }
+        }
+
+        self.ask_gcs_config(bucket_base_url, credentials_file)
+    }
+
+    fn handle_create_gcs_bucket(
+        &self,
+        project_ids: Vec<String>,
+        project_id: Option<String>,
+        bucket_name: Option<String>,
+        location: Option<String>,
+        credentials_file: Option<String>,
+    ) -> ProofStorageConfig {
+        let project_id = project_id.unwrap_or_else(|| {
+            if project_ids.is_empty() {
+                Prompt::new(MSG_CREATE_GCS_BUCKET_PROJECT_ID_NO_PROJECTS_PROMPT).ask()
+            } else {
+                PromptSelect::new(MSG_CREATE_GCS_BUCKET_PROJECT_ID_PROMPT, project_ids).ask()
+            }
+        });
+        let bucket_name =
+            bucket_name.unwrap_or_else(|| Prompt::new(MSG_CREATE_GCS_BUCKET_NAME_PROMTP).ask());
+        let location =
+            location.unwrap_or_else(|| Prompt::new(MSG_CREATE_GCS_BUCKET_LOCATION_PROMPT).ask());
+        let credentials_file = credentials_file.unwrap_or_else(|| {
+            Prompt::new(MSG_PROOF_STORE_GCS_CREDENTIALS_FILE_PROMPT)
+                .default(DEFAULT_CREDENTIALS_FILE)
+                .ask()
+        });
+
+        ProofStorageConfig::GCSCreateBucket(ProofStorageGCSCreateBucket {
+            bucket_name,
+            location,
+            project_id,
+            credentials_file,
+        })
+    }
+
+    fn ask_gcs_config(
+        &self,
+        bucket_base_url: Option<String>,
+        credentials_file: Option<String>,
+    ) -> ProofStorageConfig {
+        let bucket_base_url = bucket_base_url
```

Comment:
Could we verify provided url?

Reply:
Is it enough to check if it starts with `gs://`?

Reply:
could we check existence or not?

Reply:
gcloud storage buckets list --project={project_id} | grep 'storage_url: {url}'